Meta/gn/secondary/Meta/Lagom/Tools/CodeGenerators/LibLocale/BUILD.gn

```diff
@@ -0,0 +1,42 @@
+import("//Meta/Lagom/Tools/lagom_tool.gni")
+
+lagom_tool("GenerateDateTimeFormatData") {
+  sources = [ "GenerateDateTimeFormatData.cpp" ]
+  deps = [
+    "//Userland/Libraries/LibMain",
+    "//Userland/Libraries/LibTimeZone",
+    "//Meta/Lagom/Tools/CodeGenerators/LibUnicode:headers",
```

Comment:
This looks kind of weird.

Usually .h and corresponding .cpp files are in the same target, and then if you dep on that target, you link in the definitions for the declarations in the .h files. A source_set() with just .h files has no link-time effect at all (headers aren't compiled into anything by themselves). I suppose this would make sense for a .h file that contains only inline functions. Is that the case here?

Reply:
This was an attempt to fix a GN check error from this construct, where other code generators in other libs include a header from LibUnicode's code generator:

https://github.com/SerenityOS/serenity/blob/36ff6187f6f5e67ba3f2085b5257ec1de10cf79b/Meta/Lagom/Tools/CodeGenerators/LibLocale/GenerateDateTimeFormatData.cpp#L7

I believe GeneratorUtil.h is entirely header only, yes. It's a lot of templates.

Reply:
I'll remove this silly header only source set until we fix the fixme in code.